### PR TITLE
PR: Update preferences dialog style to match Designs

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

- This updates the display of preferences
- Update the code a bit to simplify section name parsing
- Update the use of localization for settings titles.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

Related to https://github.com/napari/napari/issues/2449

image
image

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

@liaprins-czi ,  @napari/core-devs should we put the `Save geometry` as part of Application or Appearance?

I hid the plugins with no options, that is why plugins is not appearing. Plugin order will be added on a following PR

### CODE REVIEW

**Review by ppwadhwa — APPROVED** [review]
LGTM.  thanks, @goanpeca !  :)

**Review by tlambert03 — APPROVED** [review]
👍

File: napari/utils/settings/_defaults.py [hunk 1]
```diff
@@ -63,14 +63,17 @@ class ApplicationSettings(BaseSettings, EventedModel):
...
     )
 
     first_time: bool = True
 
     # Window state, geometry and position
     save_window_geometry: bool = Field(
-        True, description="Save window size and position."
+        True,
+        title=trans._("Save Window Geometry"),
+        description="Save window size and position.",
```
Comment (Carreau) [inline napari/utils/settings/_defaults.py:103]:
```suggestion
        description=trans._("Save window size and position."),
```

**Review by Carreau — APPROVED** [review]

File: napari/_qt/qt_main_window.py [hunk 2]
```diff
@@ -271,7 +271,7 @@ def __init__(self, viewer, *, show: bool = True):
         self._help = QLabel('')
         self._status_bar.addPermanentWidget(self._help)
 
-        self.qt_viewer.viewer.theme = SETTINGS.application.theme
+        self.qt_viewer.viewer.theme = SETTINGS.appearance.theme
```
Comment (Carreau) [inline napari/_qt/qt_main_window.py:274]: `AttributeError: 'NoneType' object has no attribute 'theme'`

File: napari/utils/settings/_tests/test_settings.py [hunk 3]
```diff
@@ -122,7 +121,7 @@ def test_custom_theme_settings(settings):
 
     # No theme registered yet, this should fail
     with pytest.raises(pydantic.error_wrappers.ValidationError):
-        settings.application.theme = custom_theme_name
+        settings.appearance.theme = custom_theme_name
```
Comment (Carreau) [inline napari/utils/settings/_tests/test_settings.py:124]: Nonetype again.

**Comment by Carreau** [comment]
`SETTINGS.application` seem to be defined in `napari/utils/settings/_manager.py` and is an instance of `ApplicationSettings` in `napari/utils/settings/_defaults.py`; so I'm not sure how you want to handle the renaming application/appearance.

**Comment by goanpeca** [comment]
>SETTINGS.application seem to be defined in napari/utils/settings/_manager.py and is an instance of ApplicationSettings in napari/utils/settings/_defaults.py; so I'm not sure how you want to handle the renaming application/appearance.

For now it should be a minor nuisance, but a separate PR will handle migrations more gracefully.

Fixing the tests.

File: napari/utils/settings/_manager.py [hunk 4]
```diff
@@ -47,8 +47,6 @@ class SettingsManager:
     _FILENAME = "settings.yaml"
     _APPNAME = "Napari"
     _APPAUTHOR = "Napari"
-    application: ApplicationSettings
-    plugin: PluginSettings
```
Comment (tlambert03) [inline napari/utils/settings/_manager.py]: I'm -1 on this removal.  the problem with this is that it makes for a really bad dev experience in an IDE. With these type hints, you get tab completion, mypy checks, linting, etc... without them, we have to go into `settings/_defaults.py` each time to remember what the name is, what type they are etc...  With this pattern we lose a lot of the really nice parts of having pydantic settings.  What's the motivation for removing?

**Review by tlambert03 — CHANGES_REQUESTED** [review]
actually @goanpeca, I just noticed the removal of the type annotations from `SettingsManager`, I added those as it was kind of painful to use `SETTINGS` in an IDE in development... so curious to hear why you want to remove them again

File: napari/utils/settings/_manager.py (hunk 4 above)
Comment (goanpeca) [reply napari/utils/settings/_manager.py]: I can add them back. No problem.

Comment (tlambert03) [reply napari/utils/settings/_manager.py]: before this change, in an IDE:

Untitled 2
Untitled 3

after this change:
Untitled

Comment (tlambert03) [reply napari/utils/settings/_manager.py]: thanks!

**Review by tlambert03 — APPROVED** [review]

File: napari/utils/settings/_manager.py (hunk 4 above)
Comment (goanpeca) [reply napari/utils/settings/_manager.py]: Fixed!

File: napari/utils/settings/_tests/test_settings.py (hunk 3 above)
Comment (goanpeca) [reply napari/utils/settings/_tests/test_settings.py:124]: Fixed

**Review by sofroniewn — APPROVED** [review]
Changes look good to me. Once the merge conflict is fixed I think we can merge. Thanks for the changes and review all!

**Comment by sofroniewn** [comment]
Love this so much! Merging now, thanks for the fixes :-)

**Pull request merged**